sci-libs/vtk/vtk-9.2.2.ebuild

```diff
@@ -0,0 +1,742 @@
+	postgres? ( ${POSTGRES_DEP} )
+	python? (
+		${PYTHON_DEPS}
+		$(python_gen_cond_dep 'mpi? ( dev-python/mpi4py[${PYTHON_USEDEP}] )')
+	)
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtopengl:5
+		dev-qt/qtquickcontrols2:5
+		dev-qt/qtsql:5
+		dev-qt/qtwidgets:5
+	)
+	qt6? (
+		dev-qt/qtbase:6[gui,opengl,sql,widgets]
+		dev-qt/qtdeclarative:6[opengl]
+		dev-qt/qtshadertools:6
+		x11-libs/libxkbcommon
+	)
+	sdl? ( media-libs/libsdl2 )
+	rendering? (
+		media-libs/glew:=
+		>=media-libs/libharu-2.4.2:=
+		virtual/opengl
+		x11-libs/gl2ps
+		x11-libs/libICE
+		x11-libs/libXcursor
+		x11-libs/libXext
+	)
+	tbb? ( dev-cpp/tbb:= )
+	tk? ( dev-lang/tk:= )
+	video_cards_nvidia? ( x11-drivers/nvidia-drivers[tools,static-libs] )
+	views? (
+		x11-libs/libICE
+		x11-libs/libXext
+	)
+	web? ( ${WEBAPP_DEPEND} )
+"
+
+DEPEND="
+	${RDEPEND}
+	dev-cpp/eigen
+	dev-cpp/nlohmann_json
+	<dev-libs/pegtl-3
+	dev-libs/utfcpp
+"
+#	test? (
+#		media-libs/glew
+#		virtual/opengl
+#		x11-libs/libXcursor
+#	)
+BDEPEND="virtual/pkgconfig"
+
+#	"${FILESDIR}"/${PN}-9.0.1-0001-fix-kepler-compute_arch-if-CUDA-toolkit-11-is-used.patch
+#	"${FILESDIR}"/${P}-adjust-to-find-binaries.patch
+#	"${FILESDIR}"/${P}-avoid-naming-collision-with-netcdf-4.9.0.patch
+#	"${FILESDIR}"/${P}-Change-or-scope-struct-names-to-avoid-conflicts.patch
+PATCHES=(
+	"${FILESDIR}"/${PN}-9.2.2-vtkGeometryFilter-add-missing-mutex-header-file.patch
+	"${FILESDIR}"/${PN}-9.2.2-VTKm-respect-user-CXXFLAGS.patch
+)
+
+DOCS=( CONTRIBUTING.md README.md )
+
+# based on default settings
+CHECKREQS_DISK_BUILD="4G"
+
+pkg_pretend() {
+	[[ ${MERGE_TYPE} != binary ]] && has openmp && tc-check-openmp
+
+	if [[ $(tc-is-gcc) && $(gcc-majorversion) = 11 ]] && use cuda ; then
+		# FIXME: better use eerror?
+		ewarn "GCC 11 is know to fail building with CUDA support in some cases."
+		ewarn "See bug #820593"
+	fi
+
+	if use doc || use examples; then
+		CHECKREQS_DISK_BUILD="7G"
+	fi
+
+	if use doc && use examples; then
+		CHECKREQS_DISK_BUILD="10G"
+	fi
+
+	if use cuda; then
+		# NOTE: This should actually equal to (number of build jobs)*7G,
+		# as any of the cuda compile tasks can take up 7G!
+		# 10.2 GiB install directory, 6.4 GiB build directory with max. USE flags
+		CHECKREQS_MEMORY="7G"
+		CHECKREQS_DISK_BUILD="14G"
+	fi
+
+	if use cuda && use doc && use examples; then
+		CHECKREQS_DISK_BUILD="18G"
+	fi
+
+	if use qt6 && use qt5; then
+		ewarn "Both qt5 and qt6 USE flags have been selected. Using qt5!"
+	fi
+
+	check-reqs_pkg_setup
+
+	use postgres && postgres_check_slot
+}
+
+pkg_setup() {
+	[[ ${MERGE_TYPE} != binary ]] && has openmp && tc-check-openmp
+
+	if [[ $(tc-is-gcc) && $(gcc-majorversion) = 11 ]] && use cuda ; then
+		# FIXME: better use eerror?
+		ewarn "GCC 11 is know to fail building with CUDA support in some cases."
+		ewarn "See bug #820593"
+	fi
+
+	if use doc || use examples; then
+		CHECKREQS_DISK_BUILD="7G"
+	fi
+
+	if use doc && examples; then
+			CHECKREQS_DISK_BUILD="10G"
+	fi
```

Comment:
The error about examples is here. I hit it as well yesterday and changed it already.